python/jiminy_py/src/jiminy_py/tree.py

```diff
@@ -173,10 +174,10 @@ def _unflatten_as(data: StructNested[Any],
     """
     data_type = type(data)
     if issubclass_mapping(data_type):  # type: ignore[arg-type]
-        return data_type({  # type: ignore[call-arg]
+        return data_type(OrderedDict({  # type: ignore[call-arg]
```

Comment:
I don't get it. Why `data` is not already an `OrderedDict` instance ? It feels wrong to first create a dict, then create an ordered dict and finally create a `data_type` instance...

Reply:
When you work with gymnasium environment you can get a gymnasium.spaces.Dict.

Reply:
Ok so the issue is actually `gymnasium.spaces.Dict` sorting keys when initialised from `dict` instead of `OrderedDict`, which fells wrong in the first place.

Reply:
Yes, so I'm not sure how best to handle this case.

Reply:
while `gymnasium<1.0` is not enforcing key sorting, `gymnasium>=1.0` does, which means that this trick is going to break at some point...

Reply:
I would be worth checking if the fallback is used at the point when running the unit tests. I guess not.

Reply:
I've applied your suggestion and ran the unit tests. The fallback is not being used during the tests.

Reply:
Perfect, I made one extra minor suggestion and we are good.